iothub/device/src/Transport/Mqtt/MqttTransportHandler.cs

```diff
@@ -312,6 +317,9 @@ public override async Task OpenAsync(CancellationToken cancellationToken)
                 await _mqttClient.ConnectAsync(_mqttClientOptions, cancellationToken).ConfigureAwait(false);
                 _mqttClient.DisconnectedAsync += HandleDisconnectionAsync;
                 _mqttClient.ApplicationMessageReceivedAsync += HandleReceivedMessageAsync;
+
+                // Create a timer to remove any expired messages. The timer would invoke callback after first hour of execution and then after every minute.
+                _twinTimeoutTimer = new Timer(checkTimeout, null, _twinResponseTimeout.Milliseconds, TimeSpan.FromMinutes(1).Milliseconds);
```

Comment:
FYI, 

```csharp
TimeSpan.FromMinutes(1).Milliseconds
```

evaluates to 0 because that field means the Milliseconds section of the broader timespan. For instance,

0:1:00:25 -> 25
(hours):(minutes):(seconds):(milliseconds)

What you are looking for here is
```csharp
TimeSpan.FromMinutes(1).TotalMilliseconds
```

which evaluates to the total number of milliseconds in a minute.